### PR TITLE
Update ISSUE_TRIAGE_PROCESS.rst

### PR DESCRIPTION
**Pull request by eladkal**
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: ISSUE_TRIAGE_PROCESS.rst [hunk 1]
```diff
@@ -151,11 +154,7 @@ to be able to reproduce the issue. Typically, this may require a response to the
 issue creator asking for more information, with the issue then being tagged with
 the label ``pending-response``.
 Also, during this stage, additional labels may be added to the issue to help
-classification and triage, such as ``reported_version`` and ``area``.
-
-Occasionally an issue may require a larger discussion among the Airflow PMC or
-the developer mailing list. This issue may then be tagged with the
-``needs:discussion`` label.
+classification and triage, such as ``affected_version`` and ``area``.
```
Comment (eladkal) [inline ISSUE_TRIAGE_PROCESS.rst:157]: I'll do the rename `reported_version` -> `affected_version` after merge

**Review by auvipy — APPROVED** [review]

File: ISSUE_TRIAGE_PROCESS.rst (hunk 1 above)
Comment (o-nikolas) [reply ISSUE_TRIAGE_PROCESS.rst:157]: We should decide (and then document here) if this label will be used to tag _all_ affected versions, or just the latest/most current affected version. The former seems like it'd be interesting data but lots of paperwork, the latter seems to be what we're most interested in (knowing how current an issue is). What do folks think?

File: ISSUE_TRIAGE_PROCESS.rst [hunk 2]
```diff
@@ -206,3 +204,11 @@ following situations:
...
 * In some cases, the original creator realizes that the issue was incorrectly reported and then marks it as ``invalid``. Also, a committer could mark it as ``invalid`` if the issue being reported is for an unsupported operation or environment.
 * In some cases, the issue may be legitimate, but may not be addressed in the short to medium term based on current project priorities or because this will be irrelevant because of an upcoming change. The committer could mark this as ``wontfix`` to set expectations that it won't be directly addressed in the near term.
+
+**GitHub Discussions**
+
+Issues should represent clear feature requests which can/should be implemented. If the idea is vague or can be solved with easier steps
+we normally convert such issues to category ideas in discussion.
+Issues that seems more like support requests are also converted to discussion.
+We use judgment about what issue to convert to discussion, it's best to always clarify with comment why the issue is converted.
+Note that we can always convert discussion back to issue.
```
Comment (o-nikolas) [inline ISSUE_TRIAGE_PROCESS.rst]:
```suggestion
Issues should represent clear feature requests which can/should be implemented. If the idea is vague or can be solved with easier steps
we normally convert such issues to Discussions in the Ideas category.
Issues that seems more like support requests are also converted to Discussions in the Q&A category.
We use judgment about which Issues to convert to Discussions, it's best to always clarify with a comment why the Issue is being converted.
Note that we can always convert Discussions back to Issues.
```

**Review by kaxil — APPROVED** [review]

File: ISSUE_TRIAGE_PROCESS.rst (hunk 1 above)
Comment (eladkal) [reply ISSUE_TRIAGE_PROCESS.rst:157]: That could be useful if we would have release fixes to older versions as well but once we bump feature release then we don't fix past versions no matter what.
That said - the goal of labeling is to help contributors to navigate through the issues. If someone finds this helpful and it has merit maybe it's worth the overhead.

Currently we use policy of only 1 reported version label.

Comment (o-nikolas) [reply ISSUE_TRIAGE_PROCESS.rst:157]: Yeah, agreed on both counts. I honestly don't feel _too_ strongly either way. As long as we document what our policy is in this rst.

**Pull request merged**